hpcgap/lib/vecmat.gi

```diff
@@ -2338,26 +2338,19 @@ InstallMethod( Vector, "for a list of gf2 elements and a gf2 vector",
     ConvertToVectorRep(r,2);
     return r;
   end );
-InstallMethod( Randomize, "for a mutable gf2 vector",
-  [ IsGF2VectorRep and IsMutable ],
-  function( v ) 
-    local i;
-    MultVector(v,0);
-    for i in [1..Length(v)] do 
-        if Random(0,1) = 1 then v[i] := Z(2); fi;
-    od;
-    return v;
-  end );
-InstallMethod( Randomize, "for a mutable gf2 vector and a random source",
-  [ IsGF2VectorRep and IsMutable, IsRandomSource ],
-  function( v, rs ) 
+
+InstallMethodWithRandomSource( Randomize,
+    "for a random source and a mutable gf2 vector",
+    [ IsRandomSource, IsGF2VectorRep and IsMutable ],
+  function( rs, v )
     local i;
     MultVector(v,0);
     for i in [1..Length(v)] do 
         if Random(rs,0,1) = 1 then v[i] := Z(2); fi;
```

Comment:
Not relevant for this PR, but: this is something which probably would be several orders of magnitudes faster if done in C, as we could just generate random data in larger chunks (e.g. 64 bits at a time) and initialize the memory with it as needed, instead of iterating over each bit.